src/ucs/sys/sys.c

```diff
@@ -807,6 +839,53 @@ double ucs_get_cpuinfo_clock_freq(const char *header, double scale)
     return value * scale;
 }
 
+int ucs_sys_is_pci_device_present(int vendor_id, int device_id)
+{
+    static const char pci_devices_path[] = "/sys/bus/pci/devices";
+    struct dirent *entry;
+    ucs_status_t status;
+    long value;
+    int found;
+    DIR *dir;
+
+    found = 0;
+
+    dir = opendir(pci_devices_path);
+    if (dir == NULL) {
+        goto out;
+    }
+
+    while (!found) {
+        entry = readdir(dir);
+        if (entry == NULL) {
+            goto out_closedir;
+        }
+
+        if (entry->d_name[0] == '.') {
```

Comment:
why do you need this check

Reply:
to avoid doing someting like "/../<name>"